### ISSUE TITLE
Instant Search: use correct image alt text

### ISSUE THREAD

**Issue by bluefuton** [Automattic/jetpack#19273]
#### Is your feature request related to a problem? Please describe.
When uploading an image in WordPress' media library, it is possible to specify the alt text for it. We currently use the post title as alt text in search results, but it would be better to use the alt text the user has specified wherever it is available.

![Image-attachment-details](https://user-images.githubusercontent.com/17325/112246189-2013bb80-8cb7-11eb-98c8-aded52727921.png)

#### Describe the solution you'd like
The ElasticSearch API returns `image.alt` in the response and we use that for image alt text in the application.

cc @gibrown

**Comment by zeldman** [comment]
>  it would be better to use the alt text the user has specified wherever it is available.

 
 This times a million. Thank you!

**Comment by bluefuton** [comment]
Asked @gibrown about implementing this, and he reports:

> "We definitely don’t have it in a separate field (in ElasticSearch). We may be matching against it but my guess is that you want to actually have it in a field. If you can figure out how to extract the info then indexing it should be easy. We currently do extraction via class.wpcom-media-meta-extractor.php."

**Comment by mdbitz** [comment]
Notes

Image Alt Text is saved to the meta '_wp_attachment_image_alt'. This meta is part of the allow list for post meta and synced to the WP.com Cache. At this time the meta is not indexed to the jetpack-search elastic index which is why it isn't returned and available for use.

Should '_wp_attachment_image_alt' be moved from unindexed_postmeta to public_postmeta?

**Comment by mdbitz** [comment]
In Progress :: There are some nuances with ensuring the change doesn't impact other indexes as well as not including content multiple times in all_content that are being discussed/worked through.

**Comment by mdbitz** [comment]
Pending :: Support for alt_text has been enabled as part of searching and also as a separate field. We are planning a rebuild of the jetpack-search index that will populate this data across all sites and when complete will update here so that plugin development can follow up on usage of the new data.

**Comment by bluefuton** [comment]
Thank you for adding this @mdbitz! What is the new field named? I speculatively tried `image.alt` and `image.alt_text` from the API but no joy.

**Comment by mdbitz** [comment]
The final step here is the public_api update. postponed for the moment due to ongoing maintenance.

**Comment by mdbitz** [comment]
The field will be image.alt_text. 

Changes are in D71970 pending review from colleagues I hope to deploy it before end of week.

**Comment by bluefuton** [comment]
Fantastic. Thanks @mdbitz!

**Comment by mdbitz** [comment]
@bluefuton image.alt_text is enabled as an allowed returnable field.

Assigned to you so the Jetpack code can be updated appropriately to make use of the field. Let me know if there are any issues/concerns.

**Comment by bluefuton** [comment]
Working on this in https://github.com/Automattic/jetpack/pull/22295/.

@mdbitz I noticed that when I remove alt text from an existing image, that change is not reflected in the index (changed it 30+ mins ago)? The blog ID in question is 160302922. The original alt text was "brown wooden framed glass window" and I changed it to be empty.

**Comment by mdbitz** [comment]
Looking at the indexing logic when the details of the image change the cache db is updated, however there are no triggers to update the index for posts that have the image as a featured image.

Will need to review further the impact of adding such a trigger.
